tfx/experimental/pipeline_testing/dummy_executor.py

```diff
@@ -0,0 +1,65 @@
+# Lint as: python2, python3
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base Dummy Executor"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import absl
+from distutils import dir_util
+import os
+from typing import Any, Dict, List, Text, Optional
+
+import tensorflow as tf
+from tensorflow.python.lib.io import file_io  # pylint: disable=g-direct-tensorflow-import
+from tfx import types
+from tfx.components.base import base_executor
+
+class BaseDummyExecutor(base_executor.BaseExecutor):
+  """TFX base dummy executor."""
+  def __init__(self,
+               component_id: Text,
+               record_dir: Text,
```

Comment:
I'd like to suggest renaming record_dir with test_data_dir.
Some user dummy executors won't use recorded data.